cpp/models/ode_secirvvs/model.h

```diff
@@ -543,21 +543,45 @@ class Simulation : public BaseT
     {
     }
 
-    void apply_b161(double t)
+    using ParamT = CustomIndexArray<UncertainValue, AgeGroup>;
+
+    /**
+    * @brief Applies the effect of a new variant of a disease to the transmission probability of the model.
+    * 
+    * This function adjusts the transmission probability of the disease for each age group based on the share of the new variant.
+    * The share of the new variant is calculated based on the time `t` and the start day of the new variant.
+    * The transmission probability is then updated for each age group in the model.
+    * 
+    * Based on Equation (35) and (36) in doi.org/10.1371/journal.pcbi.1010054
+    * 
+    * @param [in] t The current time.
+    * @param [in] base_infectiousness The base infectiousness of the old variant for each age group.
+    */
+
+    void apply_variant(const double t, const ParamT base_infectiousness)
     {
+        auto start_day             = this->get_model().parameters.template get<StartDay>();
+        auto start_day_new_variant = this->get_model().parameters.template get<StartDayNewVariant>();
 
-        auto start_day   = this->get_model().parameters.template get<StartDay>();
-        auto b161_growth = (start_day - get_day_in_year(Date(2021, 6, 6))) * 0.1666667;
-        // 2 equal to the share of the delta variant on June 6
-        double share_new_variant = std::min(1.0, pow(2, t * 0.1666667 + b161_growth) * 0.01);
-        size_t num_groups        = (size_t)this->get_model().parameters.get_num_groups();
-        for (size_t i = 0; i < num_groups; ++i) {
-            double new_transmission =
-                (1 - share_new_variant) *
-                    this->get_model().parameters.template get<BaseInfectiousnessB117>()[(AgeGroup)i] +
-                share_new_variant * this->get_model().parameters.template get<BaseInfectiousnessB161>()[(AgeGroup)i];
-            this->get_model().parameters.template get<TransmissionProbabilityOnContact>()[(AgeGroup)i] =
-                new_transmission;
+        if (start_day + t >= start_day_new_variant) {
+            double share_start_day = (start_day - start_day_new_variant) * 1. / 7;
+            double days_variant    = t;
+
+            // if the start day of the new variant is in the future, we need to adjust the share of the new variant
+            if (start_day < start_day_new_variant) {
+                share_start_day = 0;
+                days_variant    = t - (start_day_new_variant - start_day);
+            }
+            double share_new_variant = std::min(1.0, pow(2, days_variant * 1. / 7 + share_start_day) * 0.01);
```

Comment:
This can be simplified significantly:

```suggestion
            const double days_variant = t - (start_day_new_variant - start_day);
            const double share_new_variant = std::min(1.0, 0.01 * pow(2, (1. / 7) * days_variant));
```